ape_safe/client.py

```diff
@@ -121,7 +155,79 @@ class ExecutedTxData(UnexecutedTxData):
 SafeApiTxData = Union[ExecutedTxData, UnexecutedTxData]
 
 
-class SafeClient:
+class BaseSafeClient(ABC):
+    @property
+    @abstractmethod
+    def safe_details(self) -> SafeDetails:
+        ...
+
+    @abstractmethod
+    def get_next_nonce(self) -> int:
+        ...
+
+    @abstractmethod
+    def _all_transactions(self) -> Iterator[SafeApiTxData]:
+        ...
+
+    def get_transactions(
+        self,
+        confirmed: Optional[bool] = None,
+        starting_nonce: int = 0,
+        filter_by_ids: Optional[Set[SafeTxID]] = None,
+        filter_by_missing_signers: Optional[Set[AddressType]] = None,
+    ) -> Iterator[SafeApiTxData]:
+        """
+        confirmed: Confirmed if True, not confirmed if False, both if None
+        """
+        next_nonce = self.get_next_nonce()
+
+        for txn in self._all_transactions():
+            if txn.nonce < starting_nonce:
+                break  # NOTE: order is largest nonce to smallest, so safe to break here
+
+            is_confirmed = len(txn.confirmations) >= txn.confirmationsRequired
```

Comment:
**todo / question**: Can we make `confirmationsRequired` snake_case` at the source? If we are for some reason unable to, disregard. Otherwise I would like to do this before people start using the plugin so we don't have to break it later.

Reply:
the EIP712 object uses the named fields for serializing and signing

just keep that in mind (otherwise can use pydantic field aliasing)

Reply:
let's save this for after this PR

Reply:
did it!